management-api-server/src/main/java/com/datastax/mgmtapi/Cli.java

```diff
@@ -436,18 +452,85 @@ void preflightChecks() {
     checkUnixSocket();
   }
 
-  private NettyJaxrsServer startHTTPService(String hostname, int port) throws SSLException {
+  @VisibleForTesting
+  void createSSLContext() throws SSLException {
+    this.sslContext =
+        SslContextBuilder.forServer(tlsCert, tlsKey)
+            .trustManager(tlsCaCert)
+            .clientAuth(ClientAuth.REQUIRE)
+            .ciphers(null, IdentityCipherSuiteFilter.INSTANCE)
+            .build();
+  }
+
+  @VisibleForTesting
+  void createSSLWatcher() throws IOException {
+    // Watch for tls_cert_file, tls_key_file and tls_ca_cert_file, add all their directories to
+    // Filesystem Watcher
+    WatchService watchService = FileSystems.getDefault().newWatchService();
+    java.nio.file.Path tlsCertParent = tlsCert.toPath().getParent();
+    java.nio.file.Path tlsKeyParent = tlsKey.toPath().getParent();
+    java.nio.file.Path tlsCaCertParent = tlsCaCert.toPath().getParent();
+
+    tlsCertParent.register(
+        watchService,
+        StandardWatchEventKinds.ENTRY_CREATE,
+        StandardWatchEventKinds.ENTRY_DELETE,
+        StandardWatchEventKinds.ENTRY_MODIFY);
+    tlsKeyParent.register(
+        watchService,
+        StandardWatchEventKinds.ENTRY_CREATE,
+        StandardWatchEventKinds.ENTRY_DELETE,
+        StandardWatchEventKinds.ENTRY_MODIFY);
+    tlsCaCertParent.register(
+        watchService,
+        StandardWatchEventKinds.ENTRY_CREATE,
+        StandardWatchEventKinds.ENTRY_DELETE,
+        StandardWatchEventKinds.ENTRY_MODIFY);
+
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
Suggestion: Can we piggy back off any existing thread pool? I'm a little hesitant to create random single threaded executors this deep down in the application. 

Maybe it would be good to inject the executor from the `main` function in either event.

Reply:
No, since this will block that executioner, so if we piggyback some existing ones, we need to ensure there's an existing single thread there always reserved for this one.

Reply:
Ok fair, so I'll amend this suggestion to be: can we find where any other thread pools are created and ensure that any thread pools we're creating are created in the same place.